Refactoring related to Selenium 4 API deprecations

quarckster opened the pull request:
- Added type annotations
- `webdriver_options` are converted to `selenium.webdriver.ChromeOptions` or `selenium.webdriver.FirefoxOptions`
- `BrowserFactory` passes only allowed arguments to the webdriver
- Tests use `podman` in a subprocess to run a container with selenium
- Added `test_session`
- Selenium 4.0 is minimal version
- `setuptools` is replaced with `hatch`
- `setup.cfg` is replaced with `pyproject.toml`
- `.pre-commit-config.yaml` updated
- `flake8` config is moved to `pre-commit`
- `.isort.cfg` is deleted
- Updated Github Actions config

RonnyPfannschmidt commented on src/webdriver_kaifuku/__init__.py:
```diff
@@ -105,32 +68,14 @@ def create(self):
 class WharfFactory(BrowserFactory):
-    wharf = attr.ib()
-
-    DEFAULT_WHARF_CHROME_OPT_ARGS = ["--no-sandbox"]
```
Does this affect cfme?

quarckster replied:
I'm not aware if cfme is still used. Anyway, I would drop wharf support at all.

RonnyPfannschmidt replied:
drop is ok

RonnyPfannschmidt reviewed (commented):
this is a massive squashed change that takes time to review, so next week when im back from pto

quarckster commented:
> this is a massive squashed change that takes time to review, so next week when im back from pto

I'll split it into meaningful commits.

Commit:
Refactored BrowserManager:
- added type annotations
- webdriver_options are converted to selenium.webdriver.ChromeOptions or selenium.webdriver.FirefoxOptions
- BrowserFactory passes only allowed arguments to the webdriver

Commit:
- Removed docker from the tests
- Tests use podman in subprocess to run a container with selenium
- Added test_session

Commit:
- Python 3.8 is minimal version
- Selenium 4.0 is minimal version
- setuptools is replaced with hatch
- setup.cfg is replaced with pyproject.toml
- .pre-commit-config.yaml updated
- flake8 config is moved to pre-commit
- .isort.cfg is deleted

Commit:
Updated Github Actions config

RonnyPfannschmidt commented on testing/conftest.py:
```diff
@@ -1,120 +1,77 @@
-WHARF_URL = "http://localhost:4899"
-WHARF_IMAGE = "cfmeqe/webdriver-wharf"
-BROWSER_IMAGE = "cfmeqe/cfme_sel_stable"
+BROWSER_IMAGE = "quay.io/redhatqe/selenium-standalone:latest"
```
followup note - lets make this something the integration has to supplement
